mat/matrix_test.go

```diff
@@ -528,6 +528,31 @@ func TestNormZero(t *testing.T) {
 	}
 }
 
+func TestRank(t *testing.T) {
```

Comment:
Please make this a table-driven test like
```
tests := []struct{
    a    *Dense
    eps  float64
    want int
} {
    // ...
}

for i, test := range tests {
    // ...
}
```
and add more test cases: wide, tall, square matrices of various sizes including trivial 1x1, 1x2, ... sizes, with full and deficient rank, diagonal matrices with a small element and tolerance that is smaller and larger that this element (so that the rank is different depending on epsilon), .... Each test case can also additionally test whether `Rank(a, .) == Rank(a.T(), .)`.

Reply:
I make it table driven. Please wait for additional test cases.